bmq-sdk/src/main/java/com/bloomberg/bmq/SessionOptions.java

```diff
@@ -408,6 +408,28 @@ private Builder() {
             hostHealthMonitor = null;
         }
 
+        /**
+         * Copies another 'SessionOptions' into this builder.
+         *
+         * @param options specifies options which is copied into the builder
+         * @return Builder this object
+         */
+        public Builder copyFrom(SessionOptions options) {
```

Comment:
We have `merge` for `QueueOptions` builder but `copyFrom` for `SessionOptions` builder.
https://github.com/bloomberg/blazingmq-sdk-java/blob/584014e290b28b0b42692f2e2832cf59e5f3e7dd/bmq-sdk/src/main/java/com/bloomberg/bmq/QueueOptions.java#L304

My concern is that the API is not consistent and a bit contrintuitive.
`copyFrom` for `SessionOptions` could be modeled by `merge` used on an empty (just built) builder. It is done this way for `QueueOptions`

Reply:
I have the same concern.
The problem is that `QueueOptions` class uses `Optional<T>` for its fields while `SessionOptions` uses primitives, `Duration` etc.

For this reason `QueueOptions::merge` copies only fields set by user while for `SessionOptions` we need to copy all fields (this was requested by another team).
Besides, `merge` makes sense for `QueueOptions` in order to accumulate options (only fields set by user) from previous queue configure requests. But it doesn't make sense (at least for now) for `SessionOptions` because it's used only when the session is started.

I see the following options:
1. Convert `SessionOptions` (and its `Builder`) fields to `Optional<T>`, add `merge` method which may be called on empty builder to model `copyFrom`
2. Add `SessionOptions::build(SessionOptions)` overload which takes existing options and initiates the builder

Option 1 requires more work to do (including testing) and I'm not sure this should be done now.
Option 2 requires less work to do. Besides, later if necessary, we may add `SessionOptions::Builder::merge()` and `QueueOptions::build(QueueOptions)` so both classes become consistent.

So I suggest to go with Option 2 for now. What do you think?

Reply:
@678098 implemented option 2